internal/app/machined/pkg/runtime/v1alpha1/platform/metal/url.go

```diff
@@ -119,7 +120,9 @@ func getResource[T resource.Resource](ctx context.Context, r state.State, namesp
 		select {
 		case <-watchCtx.Done():
 			err := fmt.Errorf("failed to determine %s of %s: %w", valName, typ, watchCtx.Err())
-			err = fmt.Errorf("%s; %w", err.Error(), watchErr)
+			if watchErr != nil {
+				err = multierror.Append(watchErr, err)
```

Comment:
why `watchErr` if we return `err` below?

Reply:
Redone here https://github.com/siderolabs/talos/pull/6605